packages/synapse-react-client/src/components/EntityTreeTable/components/ColumnHeaders.test.tsx

```diff
@@ -0,0 +1,58 @@
+import { render, screen } from '@testing-library/react'
+import {
+  NameColumnHeader,
+  CreatedOnColumnHeader,
+  ModifiedOnColumnHeader,
+} from './ColumnHeaders'
+
+// Mock ColumnHeader component
+vi.mock('../../TanStackTable/ColumnHeader', () => ({
+  default: vi.fn(({ title, ...props }) => (
+    <div data-testid="column-header" data-title={title}>
+      {title}
+    </div>
+  )),
+}))
+
+const mockHeaderContext = {
+  table: {} as any,
+  header: {} as any,
+  column: {} as any,
+}
+
+describe('ColumnHeaders', () => {
+  it('should render NameColumnHeader with correct title', () => {
+    render(<NameColumnHeader {...mockHeaderContext} />)
+
+    const header = screen.getByTestId('column-header')
+    expect(header).toHaveAttribute('data-title', 'Name')
+    expect(screen.getByText('Name')).toBeInTheDocument()
```

Comment:
All of these are basically testing the mock. I would say either use the real `ColumnHeader` component to make this an integration test, or change these assertions to

```ts
expect(MockColumnHeader).toHaveBeenRenderedWithProps(expect.objectContaining({title: "Name"}))
```

Reply:
I guess the one thing my suggestion doesn't capture, that this does, is that it ensures that the mock component is displayed on the screen (a function component could be called but not displayed). So maybe this is OK.

Reply:
Updated toHaveBeenRenderedWithProps to optionally accept a testId which it attempts to find